### ISSUE TITLE
All static files copied ontop of eachother

### ISSUE THREAD

**Issue by dns2utf8** [getpelican/pelican#2444]
Hi all

I upgraded to 4.0.0 today and noticed that all my static files are processed as "draft":
```
...
-> Copying /home/user/website/content/mindtuxmap/img/lang/ch.png to drafts/pages/.html
-> Copying /home/user/website/content/images/breeze_viewpdf.svg to drafts/pages/.html
...
```

Do I have to configure something special? It worked like a charm with 3.7.1.

Cheers

**Comment by dns2utf8** [comment]
I noticed, I can include png's or pdf's if I use `{attach}` instead of `{static}`.
I was under the impression that this should only affect the path on the generated site not the resolution itself, is this a bug?

Best

**Comment by justinmayer** [comment]
Hi Stefan. I believe @oulenz, and maybe @mosra, made some improvements for Pelican 4.0 that might relate to the issue you described. Perhaps they might have an idea what's causing that.

Keep in mind that they may need more information than you've provided. As mentioned in the [How to Get Help](http://docs.getpelican.com/en/stable/contribute.html#how-to-get-help) section of the documentation, it's hard to troubleshoot without knowing more about your environment (settings file contents, et cetera).

**Comment by oulenz** [comment]
I am traveling without access to a computer, tomorrow evening is the
earliest I can look at this. I wonder whether it might be because the
Static and Page types inherit from each other.

Den tor. 15. nov. 2018, 13.40 skrev Justin Mayer <notifications@github.com:

**Comment by dns2utf8** [comment]
Okay, what do you need to know? I am building with a gitlab runner inside a debian docker image.
I will try to use the git version tomorrow.

Currently I solved it by pinning the version to 3.7.1 for the CI.
So it is not urgent. @oulenz if you are on vacation, please enjoy it.

**Comment by avaris** [comment]
A repo that reproduces the problem would be ideal.

**Comment by oulenz** [comment]
@dns2utf8 My theory is that your settings include

```python
DEFAULT_METADATA = {
    'status': 'draft',
}
```

...or something similar to the effect that your pages have default draft status. Is this correct?

If yes, the problem is that because the `Static` class inherits from `Page`, your static content takes the metadata values of a draft page. I've written a PR that fixes that.

Otherwise we have to keep looking, but the PR is probably still pertinent.

**Comment by bryanbrattlof** [comment]
I think @oulenz has it.

I've just ran into this issue myself, and it does look like the static content is being handed over to the draft page.

With `PATH` and `DEFAULT_METADATA` set.
```python
PATH = 'content'
DEFAULT_METADATA = {                                                                                                                                                                                   
    'status': 'draft',                                                                                                                                                                                 
}
```
I get the following:
```bash
WARNING: A problem occurred copying file 'content/static/img/logo.png' to '/index.html'; PermissionError: [Errno 13] Permission denied: '/index.html'
```
Removing `DEFAULT_METADATA` gives the correct result back.
```bash
-> Copying 'content/static/img/logo.png' to 'static/img/logo.png'
```

**Comment by justinmayer** [comment]
Thanks to @oulenz, this has ostensibly been fixed. I'll try to release a Pelican 4.0.1 soon containing this and any other fixes that are ready by that time.

**Comment by dns2utf8** [comment]
Hi all

Yes I have those settings:
```
...
PATH = 'content'
STATIC_PATHS = [
    'images',
    'extra',
    'mindtuxmap',
    'slides',
]
DEFAULT_METADATA = {
    'status': 'draft',
    'lang': 'en',
}
...
```

Sorry for the late reply. If it fixes itself with 4.0.1 that would be great. I will try again after [RustFest](https://rome.rustfest.eu).
Thank you for fixing it.

Cheers,
Stefan

**Comment by dns2utf8** [comment]
On another note, I tried the git version and it works perfectly.

**Comment by oulenz** [comment]
Glad to hear that! :-)

**Comment by justinmayer** [comment]
Fix for this has been included in Pelican 4.0.1, which I just released. Enjoy! :rocket:
